h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -102,29 +108,37 @@ public static DMatrix convertFrameToDMatrix(Key<DataInfo> dataInfoKey,
 
             // truly sparse matrix - no categoricals
             // collect all nonzeros column by column (in parallel), then stitch together into final data structures
+            Vec.Reader w = weight == null ? null : weightVector.new Reader();
             if (csc) {
                 trainMat = csc(f, chunks, w, f.vec(response).new Reader(), nRows, di, resp, weights);
             } else {
+                Vec.Reader[] vecs = new Vec.Reader[f.numCols()];
+                for (int i = 0; i < vecs.length; ++i) {
+                    vecs[i] = f.vec(i).new Reader();
+                }
                 trainMat = csr(f, chunks, vecs, w, f.vec(response).new Reader(), nRows, di, resp, weights);
```

Comment:
`nrows` is unused parameter in `csr` method. Not introduced in this PR, but pointing it out. It was always this way and I forgot to remove it in my last PR. Cosmetic detail.

Reply:
If it was like that - lets keep it and do a cleanup later. The same is for the folds parameter